libr/anal/fcn.c

```diff
@@ -2301,13 +2301,13 @@ R_API ut32 r_anal_fcn_cost(RAnal *anal, RAnalFunction *fcn) {
 		return 0;
 	}
 	r_list_foreach (fcn->bbs, iter, bb) {
-		RAnalOp op;
+		RAnalOp op = R_EMPTY;
 		ut64 at, end = bb->addr + bb->size;
 		ut8 *buf = malloc (bb->size);
 		(void)anal->iob.read_at (anal->iob.io, bb->addr, (ut8 *) buf, bb->size);
 		int idx = 0;
 		for (at = bb->addr; at < end;) {
-			memset (&op, 0, sizeof (op));
+			// memset (&op, 0, sizeof (op)); // Not needed, op = R_EMPTY
```

Comment:
So, IMHO this shouldn't be needed at all ever, because `r_anal_op` should set all the necessary fields. However, setting `op = R_EMPTY` outside the loop is not the same as zeroing it at each iteration.

That said, if you want to remove the line, just remove it instead of uncommenting please.

Reply:
You're absolutely right, I did a search for all the `RAnalOp`s defined on the stack, and removed the following `memset`s where appropriate - didn't notice this one is in a loop. I will restore it, as well as remove the other `memset`s instead of commenting them out.